Assets/Scripts/Items/SpeedUpItem.cs

```diff
@@ -6,11 +6,27 @@ public class SpeedUpItem : MonoBehaviour
 {
     public float speedUpAmount;
 
+    private bool isSpawned = false;
+    private int row = -1;
+    private int col = -1;
+
+    void Update() {
+        if (isSpawned && !Level.GetSingleton().HasLilypadAt(row, col)) {
+            Destroy(gameObject);
+        }
+    }
+
     void OnTriggerEnter2D(Collider2D col) {
         if (col.CompareTag("Player")) {
             col.GetComponent<Frog>().speed += speedUpAmount;
```

Comment:
I don't think the speed boost amount of 0.3 is final, but it feels too fast currently. I was thinking of 5% increase?
But in that case, in the late game will be too small a boost. What do you think about using multiplicative instead of additive? i.e. multiply by 1.05x for every boost instead of +0.3

Reply:
Yup, just tested 1.05f factor and its good.